core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala

```diff
@@ -561,6 +566,44 @@ final class EventHubsConf private (private val connectionStr: String)
             s"${PartitionPreferredLocationStrategy.values.mkString(",")}"))
   }
 
+  /**
+   * Use AAD auth to connect eventhubs instead of connection string.
+   * More info about this: https://docs.microsoft.com/en-us/azure/event-hubs/authorize-access-azure-active-directory
+   * Default: [[false]]
+   * @return the updated [[EventHubsConf]] instance
+   */
+  def setUseAadAuth(b: Boolean): EventHubsConf = {
+    set(UseAadAuthKey, b)
+  }
+
+  def useAadAuth: Boolean = {
+    self.get(UseAadAuthKey).getOrElse(DefaultUseAadAuth).toBoolean
+  }
+
+  def setAadAuthCallback(callback: AadAuthenticationCallback): EventHubsConf = {
+    set(AadAuthCallbackKey, callback.getClass.getName)
+  }
+
+  def aadAuthCallback(): Option[AadAuthenticationCallback] = {
+    self.get(AadAuthCallbackKey) map (className => {
+      Class.forName(className).newInstance().asInstanceOf[AadAuthenticationCallback]
+    })
+  }
+
+  /**
+   * Set the tenant id if [[AadAuthKey]] is Certificate or Password
```

Comment:
Both the callback class and the tenantID is set in EventHubClient.createWithAzureActiveDirectory. I'd still prefer to keep it here as:
1) from logic, they're the same level instead of one from another. If someone has experience of writing AAD auth implementation, they would expect to set the tenant id in the out layer instead of the callback. :)
2) the "authority" parameter in the callback function(acquireToken) actually is a right value, we should not override or change it there, the right way to do is to get tenant id from authority.
3) I still prefer to keep trait clean and do not include any data in it.